internal/pubsub/gcppubsub/gcppubsub.go

```diff
@@ -74,6 +74,17 @@ func (t *topic) IsRetryable(error) bool {
 	return false
 }
 
+// As implements driver.Topic.As.
```

Comment:
Please document the types supported for each `As` function in the package godoc; see [gcsblob](https://github.com/google/go-cloud/blob/master/blob/gcsblob/gcsblob.go#L28) for an example.

Reply:
Why wouldn't the documentation for what `As` does go right above `As`?

Reply:
Because the struct you are documenting here is not exported, and therefore it won't show up in godoc at all.

Reply:
For the rabbit test, I'm not sure how to check that the connection pointer has the right value anymore, without making `pubsub.Topic.driver` public.

Reply:
Look at the generated godoc for the package (e.g., https://godoc.org/github.com/google/go-cloud/blob/s3blob for s3blob). If you put the documentation on `As`, it won't show up in godoc, because `As` is a function on an unexported struct. The only exported functions in driver packages are generally the constructors.

So, if you want users to see it, you need to put the documentation on the constructors or in the package docs.